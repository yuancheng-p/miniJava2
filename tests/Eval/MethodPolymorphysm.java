public class MethodPolymorphysm {

  public static void main() {

    A a = new A ();
    int i = a.foo();
    int j = a.foo(99);

  }

}


class A {

  int foo() {
    return 100;
  }

  int foo(int i) {
    int k = i;
    return k;
  }
}
