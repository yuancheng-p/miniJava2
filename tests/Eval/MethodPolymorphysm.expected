successfull parsing
start typing
check add and methods:MethodPolymorphysm:main
check add and methods:A:foo
check add and methods:A:foo
--------- env --------
A : class_env(parent=: Object)
method : foo
method : foo paramlist: int 
Boolean : class_env(parent=: Object)
Integer : class_env(parent=: Object)
MethodPolymorphysm : class_env(parent=: Object)
method : main
Object : class_env(parent=: )
String : class_env(parent=: Object)
end typing
_______________main___________________
A a = new A();
    /*
    ------ a in frame ------
    a : 0 (ref)
    ------ object in heap ------
    [A] (0) :
    */

int i = a.foo();
    /*
    ------ Call method: foo() ---------
    ------ Return: 100 ------------
    */

    /*
    ------ i in frame ------
    i : 100
    */

int j = a.foo(99);
    /*
    ------ Call method: foo() ---------
int k = i;
    /*
    ------ k in frame ------
    k : 99
    */

    ------ Return: 99 ------------
    */

    /*
    ------ j in frame ------
    j : 99
    */

